applications/GeoMechanicsApplication/custom_processes/set_parameter_field_process.cpp

```diff
@@ -0,0 +1,155 @@
+#include <utilities/mortar_utilities.h>
+
+#include "utilities/interval_utility.h"
+#include "geo_mechanics_application_variables.h"
+
+
+namespace Kratos
+{
+
+    SetParameterFieldProcess::SetParameterFieldProcess(ModelPart& rModelPart,
+                                                       const Parameters& rSettings)
+                                                            : Process(), mrModelPart(rModelPart),
+                                                            mParameters(rSettings)
+{
+    // function type: python, cpp, input
+    const Parameters default_parameters(R"(
+        {
+            "help"            : "This process applies a moving load condition belonging to a modelpart. The load moves over line elements.",
+            "model_part_name" : "please_specify_model_part_name",
+            "variable_name"   : "CUSTOM",
+            "func_type"       : "input",               
+            "function"        : "0",
+            "dataset"         : "dummy"
+        }  )"
+    );
+
+    mParameters.RecursivelyValidateAndAssignDefaults(default_parameters);
+
+}
+
+
+void SetParameterFieldProcess::SetValueAtElement(Element& rElement, const Variable<double>& rVar, const double Value)
+{
+
+    Properties& r_prop = rElement.GetProperties();
+
+    // Copies properties
+    Properties::Pointer p_new_prop = Kratos::make_shared<Properties>(r_prop);
+
+    // Adds new properties to the element
+    p_new_prop->SetValue(rVar, Value);
+    rElement.SetProperties(p_new_prop);
+
+}
+
+
+void SetParameterFieldProcess::SetParameterFieldUsingInputFunction(const Variable<double>& rVar)
+{
+    auto parameter_function = BasicGenericFunctionUtility(mParameters["function"].GetString());
+    const double current_time = this->mrModelPart.GetProcessInfo().GetValue(TIME);
+
+    for (Element& r_element : mrModelPart.Elements()) {
+
+        const auto& r_geom = r_element.GetGeometry();
+
+        // calculate parameter value at current element
+        const double val = parameter_function.CallFunction(r_geom.Center().X(), r_geom.Center().Y(), r_geom.Center().Z(), current_time, 0, 0, 0);
+
+        SetValueAtElement(r_element, rVar, val);
+
+    }
+}
+
+
+void SetParameterFieldProcess::SetParameterFieldUsingPythonFunction(const Variable<double>& rVar)
+{
+    // get new data from the data set
+    const std::string& r_dataset = mParameters["dataset"].GetString();
+    const Parameters new_data{ r_dataset };
+    const Vector& r_data_vector = new_data["values"].GetVector();
+
+    // set new data on the elements
+    IndexType i = 0;
+    for (Element& r_element : mrModelPart.Elements())
+    {
+        SetValueAtElement(r_element, rVar, r_data_vector[i]);
+        ++i;
+    }
+}
+
+
+void SetParameterFieldProcess::SetParameterFieldUsingInputJson(const Variable<double>& rVar)
+{
+    // Read json string in field parameters file, create Parameters
+    const std::string& field_file_name = mParameters["dataset"].GetString();
```

Comment:
Can we perhaps make a seperate parameter to clearly have "dataset_json_file" and "dataset". This is just for clarity in the input data. It is always better to know what to expected after the keyword in a json string so we can check for it.